src/lib/providers/remoteFsProvider.ts

```diff
@@ -94,14 +98,29 @@ export class RemoteFsProvider implements TreeDataProvider<FileLink> {
 
   changeWatch(srcPath: string, destPath: string, filesToUpload: any): void {
     const setWatch = () => {
-      this.currentWatcher = watch(getPortalId(), srcPath, destPath, {
-        mode: 'publish',
-        remove: true,
-        disableInitial: false,
-        notify: false,
-        commandOptions: {},
-        filePaths: filesToUpload,
-      });
+      const uploadingStatus = buildUploadingStatusBarItem();
+      uploadingStatus.show();
+      window.showInformationMessage('Beginning initial upload...');
+      this.currentWatcher = watch(
+        getPortalId(),
+        srcPath,
+        destPath,
+        {
+          mode: 'publish',
+          remove: true,
+          disableInitial: false,
+          notify: false,
+          commandOptions: {},
+          filePaths: filesToUpload,
+        },
+        () => {
+          uploadingStatus.dispose();
+          window.showInformationMessage(
+            `Initial upload complete! Now watching for changes.`
+          );
+          invalidateParentDirectoryCache(destPath);
```

Comment:
Why do we require this addition here?

Reply:
With the linked change to [cli-lib](https://github.com/HubSpot/hubspot-cli/pull/825) the `watch` function now has an optional argument for a callback to run after the initial upload completes. This is said callback! 

This allows for a little more UX feedback around the initial upload for the watch command, and refreshing the remote fs view once it finishes. Since its async, without the additional hook the only time we can run the cache invalidation and refresh the view would be immediately, which would almost definitely before the upload has finished, leading to inconsistent remote fs states and causing user confusion.